confd/response.go

```diff
@@ -28,6 +28,9 @@ type response struct {
 
 // newResponse based of the passed reader
 func newResponse(reader io.ReadCloser) (resp *response, err error) {
+	if reader == nil {
```

Comment:
IMHO that should be checked before calling newResponse. In other words the caller should be checking if there is a reader to create a response from.